### PR TITLE
add misc parameter to adjust context priority dynamically

### PR DESCRIPTION
**Pull request by XinfengZhang**
Signed-off-by: Carl Zhang <carl.zhang@intel.com>

### CODE REVIEW

File: va/va.h [hunk 1]
```diff
@@ -2547,6 +2549,23 @@ typedef struct _VAEncMiscParameterCustomRoundingControl
         uint32_t    value;
     }   rounding_offset_setting;
 } VAEncMiscParameterCustomRoundingControl;
+
```
Comment (dmitryermilov) [inline va/va.h:2552]: I assume it misses a written description of purpose of the structure and its usage.

File: va/va.h [hunk 2]
```diff
@@ -2547,6 +2549,23 @@ typedef struct _VAEncMiscParameterCustomRoundingControl
...
+{
+    union{
+        struct {
+            /** \brief indicate whether context priority changed */
+            uint32_t context_priority_update :1;
+            uint32_t reserved                :31;
+        } bits;
+        uint32_t value;
+    }flags;
+    /** \brief task priority adjustment dynamically */
```
Comment (dmitryermilov) [inline va/va.h:2563]: What does dynamically  mean here?

File: va/va.h (hunk 1 above)
Comment (XinfengZhang) [reply va/va.h:2552]: accept

File: va/va.h [hunk 3]
```diff
@@ -1878,7 +1878,9 @@ typedef enum
     /** \brief Buffer type used for FEI input frame level parameters */
     VAEncMiscParameterTypeFEIFrameControl = 18,
     /** \brief encode extension buffer, ect. MPEG2 Sequence extenstion data */
-    VAEncMiscParameterTypeExtensionData = 19
+    VAEncMiscParameterTypeExtensionData = 19,
+    /** \brief adjust context parameters dynamically, refer \c VAEncMiscParameterContextParameterAdjustment */
+    VAEncMiscParameterTypeContextParameterAdjust = 20
```
Comment (hongbolv) [inline va/va.h:1883]: The context priority is a general parameter for decoder, encoder, vpp.  The value name started with "VAEncMiscParameter" will cause the confusion.

Comment (XinfengZhang) [reply va/va.h:1883]: accept

File: va/va.h (hunk 2 above)
Comment (hongbolv) [reply va/va.h:2563]: It means the user can change priority per frame other than per stream.

**Review by hongbolv — APPROVED** [review]

**Review by dmitryermilov — APPROVED** [review]

**Pull request merged**